modules/nested-accordion/widgets/nested-accordion.php

```diff
@@ -799,6 +828,28 @@ protected function render() {
 			<?php echo $items_title_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
 		</div>
 		<?php
+
+		if ( $settings['faq_schema'] ) {
+			$json = [
+				'@context' => 'https://schema.org',
+				'@type' => 'FAQPage',
+				'mainEntity' => [],
+			];
+
+			foreach ( $faq_schema as $name => $text ) {
+				$json['mainEntity'][] = [
+					'@type' => 'Question',
+					'name' => wp_strip_all_tags( $name ),
+					'acceptedAnswer' => [
+						'@type' => 'Answer',
+						'text' => wp_strip_all_tags( $text ),
+					],
+				];
+			}
+		}
+		?>
+		<script type="application/ld+json"><?php echo wp_json_encode( $json ?? [] ); ?></script>
```

Comment:
```suggestion
		<script type="application/ld+json"><?php echo wp_json_encode( $json ); ?></script>
```
Maybe? Because the `$json` variable will never be empty, it is declared outside the loop.

Reply:
@louiswol94 I also expected this initially but `$json` is not declared anywhere outside a loop. I've specified a fallback empty value which is functionally equivalent to adding an empty `$json` array outside of the loop.

Reply:
The issue was that the conditional above this would not execute if the faq schema control was in it's default state (off) resulting in `$json` having no value in some cases.

![image](https://github.com/elementor/elementor/assets/130742176/8dfa72f6-3fac-462f-b41e-7a525f8476b5)

I've moved the script tag into the PHP conditional above which means it always have a value or never gets called.